spec/p2p/v0.34/transport.md

```diff
@@ -0,0 +1,223 @@
+The established raw connection is then submitted to a set of [filters](#connection-filtering),
+which can reject it.
+If the connection is not rejected, it is recorded in the table of established connections.
+
+The established raw TCP connection is then [upgraded](#connection-upgrade) into
+an authenticated secret connection.
+This procedure should ensure, in particular, that the public key of the remote peer
+matches the ID of the dialed peer, which is part of peer address provided to this method.
+In the absence of errors,
+the established secret connection (`conn.SecretConnection` type)
+and the information about the peer (`NodeInfo` record) retrieved and verified
+during the version handshake,
+are wrapped into an outbound `Peer` instance and returned to the switch.
+
+## Listen
+
+The `Listen` method produces a TCP listener instance for the provided network
+address, and spawns an `acceptPeers` routine to handle the raw connections
+accepted by the listener.
+The `NetAddress` method exports the listen address configured for the transport.
+
+The maximum number of simultaneous incoming connections accepted by the listener
+is bound to `MaxNumInboundPeer` plus the configured number of unconditional peers,
+using the `MultiplexTransportMaxIncomingConnections` option,
+in the node [initialization](https://github.com/tendermint/tendermint/blob/29c5a062d23aaef653f11195db55c45cd9e02715/node/node.go#L563).
+
+This method is called when a node is [started](https://github.com/tendermint/tendermint/blob/29c5a062d23aaef653f11195db55c45cd9e02715/node/node.go#L972).
+In case of errors, the `acceptPeers` routine is not started and the error is returned.
+
+## Accept
+
+The `Accept` method returns to the switch inbound connections established with a peer.
+It is a synchronous method, which blocks until a connection is accepted or an error occurs.
+The method returns an inbound `Peer` instance wrapping the established connection.
+
+The transport handles incoming connections in the `acceptPeers` persistent routine.
+This routine is started by the [`Listen`](#listen) method
+and accepts raw connections from a TCP listener.
+A new routine is spawned for each accepted connection.
+The raw connection is submitted to a set of [filters](#connection-filtering),
+which can reject it.
+If the connection is not rejected, it is recorded in the table of established connections.
+
+The established raw TCP connection is then [upgraded](#connection-upgrade) into
+an authenticated secret connection.
+The established secret connection (`conn.SecretConnection` type),
+the information about the peer (`NodeInfo` record) retrieved and verified
+during the version handshake,
+as well any error returned in this process are added to a queue of accepted connections.
+This queue is consumed by the `Accept` method.
+
+> Handling accepted connection asynchronously was introduced due to this issue:
+> https://github.com/tendermint/tendermint/issues/2047
+
+## Connection Filtering
```

Comment:
Do I understand correctly this is an internal API, only available to methods within the transport component? If so, it should likely be spec-ed separately (in a subsection, or in an implementation detail section, not sure) from the public APIs, to distinguish between public and private methods. If not, then I misunderstood something..

I found the details very interesting here, and relevant. What is confusing is I don't understand where and who sets the filters. It doesn't seem to be a configuration option as I understood from this spec, but rather an ABCI call resolves the filters. Thanks for confirming if my understanding is correct.

https://github.com/tendermint/tendermint/blob/fecd64737ce7d20f35e1c5705241ab11feb94cfa/node/node.go#L515-L516

Reply:
Yes, this is an ABCI call that gets executed if the [FilterPeers config flag](https://github.com/tendermint/tendermint/blob/83b7f4ad5bc77a56fdb51ec39ae13c1204dde6e0/config/config.go#L241) is set.